packages/graphql-hooks/src/useClientRequest.ts

```diff
@@ -249,15 +249,11 @@ function useClientRequest<
           })
         }
 
-        if (
-          revisedOpts.onSuccess &&
-          typeof revisedOpts.onSuccess === 'function'
-        ) {
-          revisedOpts.onSuccess(result, revisedOperation.variables)
-        } else {
-          if (revisedOpts.onSuccess) {
+        if (revisedOpts.onSuccess) {
+          if (typeof revisedOpts.onSuccess !== 'function') {
             throw new Error('options.onSuccess must be a function')
           }
+          revisedOpts.onSuccess(result.data, revisedOperation.variables)
```

Comment:
IMO is just a bug on the new implementation rather than a breaking change.
If you take a look here:
https://github.com/nearform/graphql-hooks/blob/91cfe6c106df3b3b49c827b7f4f1435b46dc74a8/packages/graphql-hooks/src/types/common-types.ts#L174, you can see that the `.data` field is returned: https://github.com/nearform/graphql-hooks/blob/32d2d5246539529fbc673cab126cb1b92e67ae1c/packages/graphql-hooks/src/useClientRequest.ts#L229

Reply:
this is a breaking change because it changes the signature and semantics of the onSuccess callback. whether the previous implementation was right or wrong is not relevant, it is still a breaking change because whoever was using it until now was getting the full result, with this change they will be getting only the data property